File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int zero = 0;
        int correct = 0;

        for(int i = 0; i < lottos.length; i++){
            if(lottos[i] == 0){
                zero++;
                continue;
            }
            for(int j = 0; j < win_nums.length; j++){
                if(lottos[i] == win_nums[j]){
                    correct++;
                    break;
                }
            }
        }
        int[] answer = new int[2];

        answer[0] = correct + zero;
        answer[1] = correct;

        for(int i = 0; i < answer.length; i++){
            switch (answer[i]){
                case 6:
                    answer[i] = 1;
                    break;
                case 5:
                    answer[i] = 2;
                    break;
                case 4:
                    answer[i] = 3;
                    break;
                case 3:
                    answer[i] = 4;
                    break;
                case 2:
                    answer[i] = 5;
                    break;
                default:
                    answer[i] = 6;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.swift
import Foundation

func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var result = Array<Int>()
    var top = 0
    var low = 0
    for i in lottos {
        if i == 0 {
            top += 1
            continue
        }
        for j in win_nums {
            if i == j {
                top += 1
                low += 1
            }
        }
    }
    result.append(ranking(top))
    result.append(ranking(low))
    return result
}

func ranking(_ num: Int) -> Int {
    var n = 0
    switch num {
    case 6:
        n = 1
    case 5:
        n = 2
    case 4:
        n = 3
    case 3:
        n = 4
    case 2:
        n = 5
    default:
        n = 6
    }
    return n
}